src/render/processor.rs

```diff
@@ -67,16 +78,20 @@ impl<'a> AudioParamValues<'a> {
         Self { nodes }
     }
 
-    pub(crate) fn get_raw(&self, index: &AudioParamId) -> &AudioRenderQuantum {
-        unsafe { (*self.nodes.get(&index.into()).unwrap().as_ptr()).get_buffer() }
+    pub(crate) fn get_raw(&self, index: &AudioParamId) -> AudioRenderQuantum {
+        self.nodes
+            .get(&index.into())
+            .unwrap()
+            .borrow()
+            .get_buffer()
+            .clone()
```

Comment:
Not related to this issue but maybe we will have to review the API here, I started to sketch something for #196 and this is actually quite a blocking point

Reply:
As mentioned above, we can make this work given the current method signatures.
Maybe for now ignore the `get_raw` calls for the AudioListener, we will fix that later.
To make `pub fn get(&self, index: &AudioParamId)` work, maybe add a `bool` to `AudioRenderQuantumChannel`: `is_single_valued` to denote that only the first f32 should be used. Then you should be able to get the idea right, and later we will make it nicer/more performant. 
I hope this makes sense to you. Feel free to push some work in progress any time and we will look together

Reply:
> add a bool to AudioRenderQuantumChannel: is_single_valued to denote that only the first f32 should be used

ok that was one solution I had in mind, I will continue this way then